llama.cpp

```diff
@@ -3202,7 +3222,10 @@ static bool llm_load_tensors(
                         model.output      = ml.create_tensor(ctx, tn(LLM_TENSOR_OUTPUT,      "weight"), {n_embd, n_vocab}, backend_output);
                     }
 
-                    const uint32_t n_ff = hparams.n_ff;
+                    const uint32_t n_ff      = hparams.n_ff;
+                    const int64_t n_embd_gqa = n_embd_v_gqa;
+                    GGML_ASSERT(n_embd_gqa == n_embd);
```

Comment:
Shoudn't this be:

```suggestion
                    GGML_ASSERT(n_embd_gqa == n_embd/hparams.n_gqa());
```

Existing GQA models (LLaMAv2 70B and CodeLlama 34B) are failing

Reply:
I do not have access to CodeLlama or Llama v2 model. From what I can gather from logs (e.g. https://github.com/ggerganov/llama.cpp/issues/2828):

n_head = n_head_kv = 40
n_embd = 5120

So n_gqa = 1. So the two asserts are... the same?

Reply:
But logically yes it should be as what you suggested to ensure full backward compatibility. I applied your change to other models too.

Reply:
Actually, I think about this more. This particular assert is tautological, hence useless.

By default, `n_embd_head_v = n_embd / n_head`. So `n_embd_v_gqa = n_embd_head_v * n_head_kv = (n_embd / n_head) * n_head_kv`. That is the same as `n_embd / n_gqa` because `n_gqa = n_head / n_head_kv`.